src/blocksparse/blocksparse.jl

```diff
@@ -32,13 +32,18 @@ BlockSparse(::UndefInitializer,
             blockoffsets::BlockOffsets,
             dim::Integer; vargs...) = BlockSparse(Float64,undef,blockoffsets,dim; vargs...)
 
+copy(D::BlockSparse) = BlockSparse(copy(data(D)),
+                                   copy(blockoffsets(D)))
+
+setdata(B::BlockSparse,ndata) = BlockSparse(ndata,copy(blockoffsets(B)))
```

Comment:
So I always go back and forth about whether functions like this should copy the other fields.

In this case I would lean towards not copying the block-offset list, since functions should be pretty careful about modifying the block-offset list and should make a copy before doing so. From a practical perspective, I think modifying the block-offset list in-place is quite rare so this would lead to a lot of extra copies. And if someone wanted a copy of the block-offset list, they could do it manually.

Reply:
Good catch. I was going to raise this to you as the main, non-obvious decision to think about with this PR.

I agree we can start by not copying the block offsets and then see if it leads to any trouble. Probably most if not all sensible code can either assume a view, or copy in certain cases if needed.